### ISSUE TITLE
dockutil 2.0.0 not working on 10.8

### ISSUE THREAD

**Issue by mikep345678** [kcrawford/dockutil#17]
dockutil 2.0.0 works like a champ in my testing on a 10.9 test machine, but on 10.8 machine, regardless of command (--add, --remove, --list) receive "defaults" help text (as though defaults was called with incorrect syntax):
```
$ /usr/local/bin/dockutil -v --remove TextEdit
verbose: ('processing', '/Users/mpullen/Library/Preferences/com.apple.dock.plist')
Command line interface to a user's defaults.
Syntax:

'defaults' [-currentHost | -host <hostname>] followed by one of the following:
....
```

then hangs. Ctrl-C produces the following traceback:

```
^CTraceback (most recent call last):
  File "/usr/local/bin/dockutil", line 627, in <module>
    main()
  File "/usr/local/bin/dockutil", line 260, in main
    pl = readPlist(plist_path)
  File "/usr/local/bin/dockutil", line 395, in readPlist
    plist_string = subprocess.Popen(['plutil', '-convert', 'xml1', export_fifo, '-o', '-'], stdout=subprocess.PIPE).stdout.read()
KeyboardInterrupt
```

Noticed that Python on the 10.8 machine is Python 2.7.2, but on the 10.9 machine is Python 2.7.5...

Mike

**Comment by mosman2k** [comment]
Hi there, Im having a problem with Dockutil on my 10.8 images, although it works fine for my 10.9 and 10.10 images?
Im not great with Scripts etc. so please go easy on me, but here is the script im using (that works for 10.9/10.10 -

#!/bin/bash


# Apply the dockutil settings to current logging in user $3

/usr/bin/dockutil --remove all /Users/$3
/usr/bin/dockutil --no restart --add '/Applications/Safari.app' /Users/$3
/usr/bin/dockutil --no restart --add '/Applications/Adobe Photoshop CS6/Adobe Photoshop CS6.app' /Users/$3
/usr/bin/dockutil --no restart --add '/Applications/Adobe Illustrator CS6/Adobe Illustrator.app' /Users/$3
/usr/bin/dockutil --no restart --add '/Applications/System Preferences.app' /Users/$3
/usr/bin/dockutil --no restart --add '/Applications/Microsoft Office 2011/Microsoft Word.app' /Users/$3
/usr/bin/dockutil --no restart --add '/Applications/Microsoft Office 2011/Microsoft PowerPoint.app' /Users/$3
/usr/bin/dockutil --no restart --add '/Applications/Self Service.app' /Users/$3
/usr/bin/dockutil --no restart --add '/Applications/Preview.app' /Users/$3

Whether thats enough info im not sure, but is there something needed within the script that 10.8's need spacifically? 
Any help will be appreciated...

**Comment by kcrawford** [comment]
I think there are some known issues with 2.0 on 10.8.5.

Until those are fixed I recommend using the last 1.x version.

Kyle

**Comment by mosman2k** [comment]
Hi Kyle,

Thanks for getting back to me.

I have downloaded the previous edition of dockutil (1.1.4), obviously the script I’m using now will not work in conjunction with, so looking at my script, is there anything you could suggest updating so that it works for 10.8 with the 1.1.4 dockutil script in?

Again, sorry but my scripting skill are next to nothing and I’m not sure entirely how to adapt a script to suit my 10.8 OS and dockutil 1.1.4?

Many thanks in advance and I hope that was a stupid question and made some sort of sense.

Mark

**Comment by erikng** [comment]
Perhaps install both tools on your machines and name them dockutil-1.0 and dockutil-2.0 depending on the OS version. In your script use an if statement to look for each version and based on what is installed, run your parameters.

**Comment by kcrawford** [comment]
I don't think we will go back and fix this on 10.8 at this point.  I'm going to close this.
